streams/jvm/src/main/scala/scalaz/zio/stream/platform.scala

```diff
@@ -0,0 +1,40 @@
+package scalaz.zio.stream
+
+import java.io.{ InputStream, OutputStream }
+
+import scalaz.zio._
+import scalaz.zio.blocking._
+
+trait ZStreamPlatformSpecific {
+  def fromInputStream[R](
+    is: TaskR[R, InputStream],
+    chunkSize: Int = 4096
+  ): ZStreamChunk[R with Blocking, Throwable, Byte] =
+    ZStreamChunk {
+      ZStream.managed(is.toManaged(is => effectBlocking(is.close()).orDie)) { is =>
+        effectBlocking {
+          val buf       = Array.ofDim[Byte](chunkSize)
+          val bytesRead = is.read(buf)
+
+          if (bytesRead < 0) None
+          else if (0 < bytesRead && bytesRead < buf.size) Some(Chunk.fromArray(buf).take(buf.size))
+          else Some(Chunk.fromArray(buf))
+        }
+      }
+    }
+}
+
+trait ZSinkPlatformSpecific {
+  def fromOutputStream[R](
+    os: TaskR[R, OutputStream]
+  ): ZManaged[R with Blocking, Throwable, ZSink[R with Blocking, Throwable, Nothing, Chunk[Byte], Int]] =
+    os.toManaged(os => effectBlocking { os.flush(); os.close() }.orDie).map { os =>
+      ZSink.foldM(Task.succeed(0)) { (bytesWritten, byteChunk: Chunk[Byte]) =>
+        effectBlocking {
+          val bytes = byteChunk.toArray
+          os.write(bytes)
+          ZSink.Step.more(bytesWritten + bytes.size): ZSink.Step[Int, Nothing]
```

Comment:
Oh, why is the type annotation necessary?

Reply:
Whoops! It wasn't necessary. Got rid of it.